src/actions/config.js

```diff
@@ -40,6 +43,38 @@ export function applyDefaults(config) {
     });
 }
 
+function validateCollection(collection) {
+  const {
+    name,
+    folder,
+    files,
+    format,
+    extension,
+    frontmatter_delimiter: delimiter,
+    fields,
+  } = collection.toJS();
+
+  if (!folder && !files) {
+    throw new Error(`Unknown collection type for collection "${name}". Collections can be either Folder based or File based.`);
+  }
+  if (format && !supportedFormats.includes(format)) {
+    throw new Error(`Unknown collection format for collection "${name}". Supported formats are ${supportedFormats.join(',')}`);
+  }
+  if (!format && extension && !formatByExtension(extension)) {
+    // Cannot infer format from extension.
+    throw new Error(`Please set a format for collection "${name}". Supported formats are ${supportedFormats.join(',')}`);
+  }
+  if (delimiter && !frontmatterFormats.includes(format)) {
+    // Cannot set custom delimiter without explicit and proper frontmatter format declaration
+    throw new Error(`Please set a proper frontmatter format for collection "${name}" to use a custom delimiter. Supported frontmatter formats are yaml-frontmatter, toml-frontmatter, and json-frontmatter.`);
+  }
+  if (!!folder && !selectIdentifier(collection)) {
```

Comment:
I don't think the `!!` bool coercion here is necessary. Any value which would turn into `false` due to the coercion would also be ignored due to the `&&` which follows immediately after.